### DISCUSSION TITLE
Confuse about base.css with v3.0.0-alpha.1

### DISCUSSION THREAD

**Discussion by sinyayadynya** [tailwindlabs/tailwindcss#5678] [Help]
I'm using Tailwind within [Drupal Emulsify](https://github.com/emulsify-ds/emulsify-drupal) theme with Webpack.

With TailwindCSS 2, I was loading the base.min.css into the Drupal theme library.yml file:

`tailwind:
  css:
    theme:
      node_modules/tailwindcss/dist/base.min.css: {}`

and I had the following code within `../components/style.scss`:

```
// Tailwind
@import '~tailwindcss/base';
@import '~tailwindcss/components';

// Components
@import '00-base/**/*.scss';
@import '01-atoms/**/*.scss';
@import '02-molecules/**/*.scss';

@import '03-organisms/**/*.scss';
@import '04-templates/**/*.scss';

// Tailwind utilities
@import '~tailwindcss/utilities'
```

and in tailwind.config.js :
`module.exports = {
  content: ['./components/**/*.twig', './templates/**/*.html.twig'],
}`

Following @VicGUTT 's [advice](https://github.com/tailwindlabs/tailwindcss/discussions/5668#discussioncomment-1417767) I have now removed the `base.min.css` from the library and while I have no more error, I'm still obviously missing base.css and can't find where/how to add @​tailwind base. 

Any help most welcome!

**Comment by adamwathan** [comment]
Hmm the line `@import '~tailwindcss/base';` should be giving you `@tailwind base`, as that's all that's in that file:

https://github.com/tailwindlabs/tailwindcss/blob/master/base.css

Can you provide an actual reproduction project I can pull down and run?

**Reply by sinyayadynya** [reply]
Sure @adamwathan just published the full code here: https://github.com/sinyayadynya/melon

**Reply by adamwathan** [reply]
I can't get this to build, does this build on your machine when you run `npm run develop`? Should I be running a different command?

![image](https://user-images.githubusercontent.com/4323180/135714173-9f16753a-443a-474f-a905-15bfa36d2e9b.png)

My first thought though is to remove this:

```diff
  // Libraries
  // @import '~normalize.css/normalize';
  // @import '~breakpoint-sass/stylesheets/breakpoint';

- @​tailwind base;

  // Tailwind
  @import '~tailwindcss/base';
  @import '~tailwindcss/components';

  // Components
  @import '00-base/**/*.scss';
  @import '01-atoms/**/*.scss';
  @import '02-molecules/**/*.scss';

  @import '03-organisms/**/*.scss';
  @import '04-templates/**/*.scss';

  // Tailwind utilities
  @import '~tailwindcss/utilities';
```

You effectively have base included twice here, but not sure since I can't get it to build 🤔

**Reply by sinyayadynya** [reply]
Please try `yarn build` instead of `npm run develop`

**Reply by adamwathan** [reply]
Same issue with `yarn build`:

![image](https://user-images.githubusercontent.com/4323180/135714473-9f8ec23e-3317-4b50-90e5-705373f1b1a4.png)

There's no `breakpoint` mixin defined in any of the SCSS files which is causing the error for me:

![image](https://user-images.githubusercontent.com/4323180/135714492-37674987-d27c-45e4-814c-fd293308440e.png)

**Reply by sinyayadynya** [reply]
Strange, all breakpoints' mixin are in ./components/00-base/00-defaults/_02-breakpoints.scss and if works for me locally. 
Maybe I should publish a more basic version of that theme on Github, but need 2-3 hours to prepare it.

**Reply by sinyayadynya** [reply]
I have checked and re-uploaded the theme at https://github.com/sinyayadynya/melon. Don't know what went wrong, probably too much in hurry, sorry. Now the `yarn build` should work.

**Reply by adamwathan** [reply]
Looks like it's private now maybe because I'm getting a 404, can you double check and then I can take a look?

**Reply by sinyayadynya** [reply]
@adamwathan seems that everything is OK now. What made me think that I lost base.css is because some Emulsify bem stuff started to interfere, while somehow it wasn't before. 

Also ended using the following code for style.scss:

```
// Libraries
// @import '~normalize.css/normalize';
@import '~breakpoint-sass/stylesheets/breakpoint';

// Tailwind
@import 'tailwindcss/base';
@import 'tailwindcss/components';

// Components
@import '00-base/**/*.scss';
@import '01-atoms/**/*.scss';
@import '02-molecules/**/*.scss';

@import '03-organisms/**/*.scss';
@import '04-templates/**/*.scss';

// Tailwind Utilities
@import 'tailwindcss/utilities';
```

And now everything looks fine!

**Reply by sinyayadynya** [reply]
> Looks like it's private now maybe because I'm getting a 404, can you double check and then I can take a look?

Yep, I really was too much in hurry. Now updating it again in public: https://github.com/sinyayadynya/melon.

But as explained above, it seems that everything is super fine now. A really big thanks for caring @adamwathan and sorry for the mess.

Only noticed a small dependency issue with tailwindcss-aspect-ratio: https://github.com/tailwindlabs/tailwindcss-aspect-ratio/issues/24

This apart, I'm very much tempted to build a new Drupal base theme using Vite, Tailwind and Alpine. I liked the atomic structure of Emulsify, but most of the time do not use Storybook and all the time I have to replace most of the bem code with Tailwind in twig. Maybe do you know someone interested to collaborate on such project?

**Reply by simonswiss** [reply]
Good to hear you've got it sorted!

Repo still seems to be private – one thing I was wondering is you might get a lot of repeating (or conflicting) styles between `normalize.css/normalize` and `tailwindcss/base` (unless you opt out of Tailwind's `preflight`).

And might be worth spending some time to mirror the `breakpoint-sass/stylesheets/breakpoint` over to `theme.screens` in the Tailwind config file so everything lines up nicely.

You might find this video relevant for integrating Tailwind CSS with an existing CSS codebase:

https://www.youtube.com/watch?v=oG6XPy1t1KA
